internal/controller/todo.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/daniyalumer/todo-list-go-chi/internal/helper"
+	"github.com/daniyalumer/todo-list-go-chi/internal/models"
+	"github.com/daniyalumer/todo-list-go-chi/internal/service"
+	"github.com/go-chi/chi/v5"
+)
+
+func TodoRoutes() chi.Router {
+	r := chi.NewRouter()
+
+	r.Get("/", GetTodosHandler)
+	r.Post("/", CreateTodoHandler)
+	r.Put("/", UpdateTodoHandler)
+	r.Delete("/", DeleteTodoHandler)
+
+	return r
+}
+
+func GetTodosHandler(w http.ResponseWriter, r *http.Request) {
+	todos, err := service.ReadTodoList()
+	if err != nil {
+		helper.TodoListResponseWriter(w, "failed to read todo list", []models.Todo{}, http.StatusBadRequest)
+		return
+	}
+	helper.TodoListResponseWriter(w, "todo list read successfully", todos, http.StatusAccepted)
+}
+
+func CreateTodoHandler(w http.ResponseWriter, r *http.Request) {
+	if !helper.ParseForm(w, r) {
+		return
+	}
+
+	todoDescription := r.FormValue("description")
+	userIDStr := r.URL.Query().Get("userId")
+
+
+	userID, err := strconv.Atoi(userIDStr)
+	if err != nil {
+		http.Error(w, "Invalid ID", http.StatusBadRequest)
+		return
+	}
+
+	todo, err := service.CreateTodo(todoDescription, userID)
+	if err != nil {
+		helper.TodoResponseWriter(w, "failed to create todo item", models.Todo{}, http.StatusBadRequest)
+		return
+	}
+	helper.TodoResponseWriter(w, "created todo item successfully", todo, http.StatusAccepted)
+}
+
+func UpdateTodoHandler(w http.ResponseWriter, r *http.Request) {
+	if !helper.ParseForm(w, r) {
+		return
+	}
+
+	todoDescription := r.FormValue("description")
+	completedStr := r.FormValue("completed")
+	Completed := completedStr == "true"
+
+	IdStr := r.URL.Query().Get("id")
+	Id, err := strconv.Atoi(IdStr)
+	if err != nil {
+		http.Error(w, "Invalid ID", http.StatusBadRequest)
+		return
+	}
+
+	todo, err := service.UpdateTodo(Id, Completed, todoDescription)
+	if err != nil {
+		helper.TodoResponseWriter(w, "failed to update todo item", models.Todo{}, http.StatusBadRequest)
+		return
+	}
+	helper.TodoResponseWriter(w, "updated todo item successfully", todo, http.StatusAccepted)
```

Comment:
Please explain the reason behind mentioning the accepted status code?

Reply:
I just read on this, and the default response is 200 for the http.ResponseWriter. Thus, we will not have to define a positive response. Moreover, In this case, the usage of http.StatusAccepted is incorrect since it's for requests that are pending for approval.